pi4j-core/src/main/java/com/pi4j/platform/Platform.java

```
@@ -94,95 +96,112 @@ default <T extends Provider> T provider(Class<T> providerClass) throws ProviderN
-            if(provider.id().equalsIgnoreCase(providerId) && providerClass.isInstance(provider)){
-                return (T)provider;
+    default <T extends Provider> T provider(String providerId, Class<T> providerClass) throws ProviderNotFoundException {
+        for (Provider provider : providers().values()) {
+            if (provider.id().equalsIgnoreCase(providerId) && providerClass.isInstance(provider)) {
+                return (T) provider;
             }
         }
         throw new ProviderNotFoundException(providerId, providerClass);
     }
 
-    /** {@inheritDoc} */
+    /**
+     * <p>provider.</p>
+     *
+     * @param providerId a {@link java.lang.String} object.
+     * @param <T>
+     * @return
+     * @throws ProviderNotFoundException
+     */
     default <T extends Provider> T provider(String providerId) throws ProviderNotFoundException {
 
         // first attempt to resolve by direct unique identifier
-        if(providers().containsKey(providerId)){
-            return (T)providers().get(providerId);
+        if (providers().containsKey(providerId)) {
+            return (T) providers().get(providerId);
         }
 
         // additionally attempt to resolve the provider by its class name
         try {
             Class providerClass = Class.forName(providerId);
-            if(providerClass != null && Provider.class.isAssignableFrom(providerClass)){
+            if (providerClass != null && Provider.class.isAssignableFrom(providerClass)) {
                 // iterate over providers looking for a matching class/interface
-                for(Provider provider : providers().values()) {
+                for (Provider provider : providers().values()) {
                     if (providerClass.isInstance(provider)) {
                         return (T) provider;
                     }
                 }
             }
-        } catch (ClassNotFoundException e){}
+        } catch (ClassNotFoundException e) {
```

I know this is not part of the PR, but an empty catch? That isn't a good thing, now is it?